### DISCUSSION TITLE
Cleaning cache/pdf folder

### DISCUSSION THREAD

**Discussion by davide-alghi** [YetiForceCompany/YetiForceCRM#14148] [Feature improvement]
It would be useful to clean the cache/pdf folder: it's filled by 'Generate PDF file -> One PDF file' and by 'Send PDF WorkFlow task'.
For 'Send PDF WorkFlow task', it's not an easy job due to 'Execution schedule -> Specific date'.
The folder could get very heavy.

**Comment by mariuszkrzaczkowski** [comment]
That's why it exists

https://github.com/YetiForceCompany/YetiForceCRM/blob/ef4e83d165a37d7c7027bcbee0bc3c8079d13bf5/modules/Vtiger/crons/Cache.php#L15-L38

**Reply by davide-alghi** [reply]
ah ok, I didn't know about it
but if I set a workflow running every day with a task sending a PDF (with "Send PDF" task), it'll work also after 30 days?

**Reply by davide-alghi** [reply]
I'm replying to myself ... yes it'll work :D

**Reply by mariuszkrzaczkowski** [reply]
I improved this mechanism, see clearing the cache from the CLI 

https://user-images.githubusercontent.com/10198654/106510954-e4871d00-64cf-11eb-862f-5baea2ab6606.mp4

**Reply by davide-alghi** [reply]
great :thumbsup:

**Comment by rskrzypczak** [comment]
If I am not mistaken, there should be no PDF files in this directory. All temporary files generated there should be deleted automatically when you downloaded the PDF or sent it by e-mail.

In older versions, the system did leave files in it, but currently it shouldn't.
If there are files left in this directory for the new pdfs, it means that something went wrong, e.g. the e-mail with pdf was not sent.

**Reply by davide-alghi** [reply]
yes @rskrzypczak,
when you send by email or download, temp files are deleted. As I wrote, temp files are kept with 'Generate PDF file -> One PDF file' and 'Send PDF WorkFlow task'.
Anyway ... I'll check better on last stable version.

**Reply by rskrzypczak** [reply]
I meant all the operations that generate the Pdf file, including the two that you mentioned.
If some functionality does not delete these temporary files, treat it as a bug.

P.S.
I checked it on the current development version and it seems fine.
